### PR TITLE
added type hints to match doc strings for material class

### PR DESCRIPTION
**Pull request by shimwell**
Following on from #1992 this PR adds type hinting to another class in the Python API. This adds tpye hints to the openmc.Material() class.

I wasn't entirely sure if we want to accept os.Pathlike as well as a string for the ```from_xml``` function?

### CODE REVIEW

**Comment by shimwell** [comment]
Some methods in the ```openmc.Material``` class accept arguments of type ```openmc.Material``` this causes a circular import problem if they are used as type hints. So for now I've taken out this troublesome type hints

**Comment by shimwell** [comment]
Just to mention, some of the types have not been added to methods to avoid circular imports.

For example adding ```:openmc.VolumeCalculation``` as the type results in a circular import

```
def add_volume_information(self, volume_calc: openmc.VolumeCalculation):
```

So I've left this method as is

```
def add_volume_information(self, volume_calc):
```

There is however another way to handel this using strings[, CadQuery uses](https://github.com/CadQuery/cadquery/blob/c9d3f1e693d8d3b59054c8f10027c46a55342b22/cadquery/occ_impl/shapes.py#L1141) ```"Shape"``` instead of ```cq.Shape``` which is another way to avoid circular imports. I would be happy to do this if this is the preferred approach?

File: openmc/material.py [hunk 1]
```diff
@@ -267,7 +271,7 @@ def fissionable_mass(self):
         return density*self.volume
 
     @classmethod
-    def from_hdf5(cls, group):
+    def from_hdf5(cls, group: str):
```
[the remark this answers is not in the document]
Comment (shimwell) [reply openmc/material.py]: Ah yes good spot and sorry for the mistake

I can think of three options for this
 - add an import to get it working nicely, Something like ```import h5py``` and then ```:h5py.Group``` (I've committed this option to the PR) 
 - perhaps this is one doc string that can be skipped (avoids overhead of import)
 - alternatively I could add it as a string ```"h5py.Group"``` which tells the user the type without the import but an IDE won't pick up on the type fully.

Let me know if you want a different option or something different altogether

**Review by paulromano — APPROVED** [review]

**Comment by paulromano** [comment]
Thanks @shimwell!

**Pull request merged**